src/core/src/op/experimental_detectron_generate_proposals.cpp

```diff
@@ -45,17 +44,36 @@ bool op::v6::ExperimentalDetectronGenerateProposalsSingleImage::visit_attributes
     return true;
 }
 
-void op::v6::ExperimentalDetectronGenerateProposalsSingleImage::validate_and_infer_types() {
+void ExperimentalDetectronGenerateProposalsSingleImage::validate_and_infer_types() {
     OV_OP_SCOPE(v6_ExperimentalDetectronGenerateProposalsSingleImage_validate_and_infer_types);
 
-    std::vector<ov::PartialShape> output_shapes = {ov::PartialShape{}, ov::PartialShape{}};
-    std::vector<ov::PartialShape> input_shapes = {get_input_partial_shape(0),
-                                                  get_input_partial_shape(1),
-                                                  get_input_partial_shape(2),
-                                                  get_input_partial_shape(3)};
-    shape_infer(this, input_shapes, output_shapes);
+    const auto input_size = get_input_size();
+    auto out_et = element::dynamic;
+
+    auto input_shapes = std::vector<ov::PartialShape>();
+    input_shapes.reserve(input_size);
+
+    for (size_t i = 0; i < input_size; ++i) {
+        NODE_VALIDATION_CHECK(this,
+                              element::Type::merge(out_et, out_et, get_input_element_type(i)) &&
+                                  (out_et.is_dynamic() || out_et.is_real()),
+                              "Input[",
+                              i,
+                              "] type '",
+                              get_input_element_type(i),
+                              "' is not floating point or not same as others inputs.");
```

Comment:
As the input element types were not validated before, it's always worth to ensure that it's backward compatible change.

Reply:
Right now looks there is no issue, this can check can be easily set to less strict or removed if there will be any issue.